stix2validator/validator.py

```diff
@@ -295,42 +355,39 @@ def validate_file(fn, options=None):
         options: An instance of ``ValidationOptions``.
 
     Returns:
-        An instance of ValidationResults.
+        An instance of FileValidationResults.
 
     """
-    results = ValidationResults(fn=fn)
+    file_results = FileValidationResults(filepath=fn)
     output.info("Performing JSON schema validation on %s" % fn)
 
     if not options:
         options = ValidationOptions(files=fn)
 
     try:
         with open(fn) as instance_file:
-            instance = json.load(instance_file)
+            file_results.object_results = validate(instance_file, options)
 
-        if options.files:
-            results = validate_instance(instance, options)
-
-    except SchemaInvalidError as ex:
-        results.fatal = ValidationErrorResults(ex)
-        msg = ("File '{fn}' was schema-invalid. No further validation "
-               "will be performed.")
-        output.info(msg.format(fn=fn))
     except Exception as ex:
         if 'Expecting value' in str(ex):
             line_no = str(ex).split()[3]
-            results.fatal = ValidationErrorResults('Invalid JSON input on line'
-                                                   ' %s' % line_no)
+            file_results.fatal = ValidationErrorResults(
+                'Invalid JSON input on line %s' % line_no
+            )
         else:
-            results.fatal = ValidationErrorResults(ex)
+            file_results.fatal = ValidationErrorResults(ex)
+
         msg = ("Unexpected error occurred with file '{fn}'. No further "
                "validation will be performed: {error}")
         output.info(msg.format(fn=fn, error=str(ex)))
 
-    if results.errors or results.fatal:
-        results.is_valid = False
+    if isinstance(file_results.object_results, list):
+        file_results.is_valid = all(object_result.is_valid
+                                    for object_result in file_results.object_results)
+    else:
+        file_results.is_valid = file_results.object_results.is_valid
 
```

Comment:
Having `FileValidationResults.object_results` be sometimes a list and sometimes a single result object makes other code kinda ugly.  See `output.print_results()` and `code.get_code()` too.  Should it be always the same type, even though the validation functions can return different types?

Reply:
Yeah, this is less than ideal. Can we make "object_results" always be a list (sometimes a single item), and add "object_result" which returns the actual object in the case that there's only one (and error otherwise)?

Reply:
Done.